### PR TITLE
Configure Remove book and Add Book reducers

### PR DESCRIPTION
**Pull request by tawakuliKH**
In this module:

- [ ] I listed the books with redux toolkit
- [ ] I added Remove book functionality with redux reducers
- [ ] I added Add book functionality with redux reducerse

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (zieeco) [inline src/App.js]: - Hey, please, kindly remove this eslint disabling, it does not affect your code and it is not a good practice disabling the linter, this check helps us write clean and maintainable code.

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 import './App.css';
 // eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (zieeco) [inline src/App.js]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/App.js [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 import Navbar from './components/Navbar';
-import BookList from './components/BookList';
+// import BookList from './components/BookList';
```
Comment (zieeco) [inline src/App.js]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/App.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
+import BooksContainer from './components/pages.js/BooksContainer';
 
 function App() {
+  // const dispatch = useDispatch();
```
Comment (zieeco) [inline src/App.js]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/components/BookList.js [hunk 5]
```diff
@@ -1,44 +1,32 @@
...
-      <div className="bookTitle">
-        <div className="action"> Action</div>
-        <h3 className="title">Book 1</h3>
-        <h4 className="author">John John</h4>
-        <div className="comments-action">
-          <button type="button" className="comment">Comment</button>
-          <button type="button" className="remove">Remove</button>
-          <button type="button" className="edit">Edit</button>
+const BookList = ({
+  // eslint-disable-next-line react/prop-types
```
Comment (zieeco) [inline src/components/BookList.js:5]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/components/pages.js/BooksContainer.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (zieeco) [inline src/components/pages.js/BooksContainer.js]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/redux/books/bookSlice.js [hunk 7]
```diff
@@ -1,19 +1,39 @@
+/* eslint-disable no-unused-vars */
 // eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (zieeco) [inline src/redux/books/bookSlice.js]: - Kindly remove this commented piece of code, it makes your code not maintainable and clean.

File: src/components/AddForm.js [hunk 8]
```diff
@@ -1,13 +1,40 @@
...
-  <div className="book-form">
-    <h3 className="add-book">Add New Book</h3>
-    <input type="text" placeholder="Book Name" />
-    <select name="cars" id="cars">
-      <option value="book1">Book 1</option>
-      <option value="book2">Book 2</option>
-    </select>
-    <button className="add-button" type="button">Add Book</button>
-  </div>
+/* eslint-disable no-unused-vars */
```
Comment (zieeco) [inline src/components/AddForm.js]: - Hey, please, kindly remove this eslint disabling, it does not affect your code and it is not a good practice disabling the linter, this check helps us write clean and maintainable code.

File: src/redux/books/bookSlice.js [hunk 9]
```diff
@@ -1,19 +1,39 @@
+/* eslint-disable no-unused-vars */
```
Comment (zieeco) [inline src/redux/books/bookSlice.js]: - Hey, please, kindly remove this eslint disabling, it does not affect your code and it is not a good practice disabling the linter, this check helps us write clean and maintainable code.

File: src/components/AddForm.js [hunk 10]
```diff
@@ -1,13 +1,40 @@
...
+      </select>
+      <button
+        className="add-button"
+        type="button"
+        onClick={submitForm}
+      >
+        Add Book
+      </button>
+    </div>
+  );
```
Comment (zieeco) [inline src/components/AddForm.js]: - Hey, your forms get submitted when the input field is empty, this is not a good user experience. Please, kindly fix this to improve your ap.

File: src/redux/books/bookSlice.js [hunk 11]
```diff
@@ -1,19 +1,39 @@
...
     addBook: (state, action) => {
-      state.books = state.books.push(action.payload);
+      state.books.push(action.payload);
     },
     removeBook: (state, action) => {
-      state.books = state.books.pop(action.payload);
+      const itemId = action.payload;
+
+      state.books = state.books.filter((book) => book.item_id !== itemId);
     },
```
Comment (zieeco) [inline src/redux/books/bookSlice.js:35]: - After deleting the hardcoded books, if the user adds more books, on deleting any book, all the added books get deleted. Please, kindly fix this so all books are not deleted if a user deletes a particular book, instead the remaining should be left should they not be deleted by the user.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @tawakuliKH,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- redux well initiated ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @tawakuliKH 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**